packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts

```diff
@@ -59,16 +60,27 @@ async function createDataObject<TObj extends PureDataObject, I extends DataObjec
 
     // request mixin in
     runtimeClass = mixinRequestHandler(
-        async (request: IRequest, runtimeArg: FluidDataStoreRuntime) =>
-            (await PureDataObject.getDataObject(runtimeArg)).request(request),
-            runtimeClass);
+        async (request: IRequest, runtimeArg: FluidDataStoreRuntime) => {
+            const router = (await runtimeArg.IFluidHandle?.get() as FluidObject<IFluidRouter>).IFluidRouter;
```

Comment:
I really prefer to avoid any `as` casting as it can be very unsafe, you are asserting to the complier you know best and it will never type check even if pre-conditions change.
do something like this work?
```
            const maybeRouter: FluidObject<IFluidRouter>= await runtimeArg.IFluidHandle?.get();
            assert(maybeRouter.IFluidRouter !== undefined, "Data store runtime handle is not an IFluidRouter");
            return maybeRouter.IFluidRouter.request(request);
```

Reply:
Coming in next push